engine/gamesys/src/gamesys/components/comp_factory.cpp

```diff
@@ -231,13 +236,27 @@ namespace dmGameSystem
                     dmGameObject::ReleaseInstanceIndex(index, collection);
                 }
             }
+
+            if (properties)
+                dmGameObject::PropertyContainerDestroy(properties);
         }
         return dmGameObject::UPDATE_RESULT_OK;
     }
 
-    static dmGameObject::HPrototype GetPrototype(dmResource::HFactory factory, FactoryComponent* component)
+    static FactoryResource* CompFactoryGetResourceInternal(HFactoryComponent component)
+    {
+        return component->m_CustomResource ? component->m_CustomResource : component->m_Resource;
+    }
+
+    FactoryResource* CompFactoryGetResource(HFactoryWorld world, HFactoryComponent component)
     {
-        FactoryResource* resource = CompFactoryGetResource(component);
+        (void)world;
```

Comment:
While not technically needed, it seems nice to have a similar api from all the components in the form "CompMyTypeDoSomething(world, component)"